Packs/MISP/Integrations/MISPV3/MISPV3.py

```diff
@@ -0,0 +1,1411 @@
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='MISP.Event',
+        outputs_key_field='ID',
+        outputs=build_events_search_response(response),
+        raw_response=response
+    )
+
+
+def add_sighting(demisto_args: dict):
+    """Adds sighting to MISP attribute
+    """
+    attribute_id = demisto_args.get('id')
+    attribute_uuid = demisto_args.get('uuid')
+    sighting_type = demisto_args['type']  # mandatory arg
+    att_id = attribute_id or attribute_uuid
+    if not att_id:
+        return_error('ID or UUID not specified')
+    sighting_args = {
+        'id': attribute_id,
+        'uuid': attribute_uuid,
+        'type': SIGHTING_TO_TYPE_ID[sighting_type]
+    }
+    sigh_obj = MISPSighting()
+    sigh_obj.from_dict(**sighting_args)
+    response = PYMISP.add_sighting(sigh_obj, att_id)
+    if response.get('message'):
+        return_error(f"An error was occurred: {response.get('message')}")
+    elif response.get('Sighting'):
+        human_readable = f'Sighting \'{sighting_type}\' has been successfully added to attribute {att_id}'
+        return CommandResults(readable_output=human_readable)
+    return_error(f"An error was occurred: {json.dumps(response)}")
+
+
+def test(malicious_tag_ids, suspicious_tag_ids):
+    """
+    Test module.
+    """
+    is_tag_list_valid(malicious_tag_ids)
+    is_tag_list_valid(suspicious_tag_ids)
+    response = PYMISP._prepare_request('GET', 'servers/getPyMISPVersion.json')
+    if PYMISP._check_json_response(response):
+        return 'ok'
+    else:
+        return_error('MISP has not connected.')
+
+
+def add_events_from_feed(demisto_args: dict, use_ssl: bool, proxies: dict):
+    """Gets an OSINT feed from url and publishing them to MISP
+    urls with feeds for example: `https://www.misp-project.org/feeds/`
+    feed format must be MISP.
+    """
+    headers = {'Accept': 'application/json'}
+    url = demisto_args.get('feed')  # type: str
+    url = url[:-1] if url.endswith('/') else url
+    if PREDEFINED_FEEDS.get(url):
+        url = PREDEFINED_FEEDS[url].get('url')  # type: ignore
+    limit = demisto_args.get('limit')  # type: str
+    limit_int = int(limit) if limit.isdigit() else 0
+    osint_url = f'{url}/manifest.json'
+    not_added_counter = 0
+    try:
+        uri_list = requests.get(osint_url, verify=use_ssl, headers=headers, proxies=proxies).json()
+        events_numbers = list()  # type: List[Dict[str, int]]
+        for num, uri in enumerate(uri_list, 1):
+            req = requests.get(f'{url}/{uri}.json', verify=use_ssl, headers=headers, proxies=proxies).json()
+            e = MISPEvent()
+            e.load(req)
+            event = PYMISP.add_event(e)
+            event_data = event.get('Event')
+            if event_data and 'id' in event_data:
+                events_numbers.append({'ID': event_data['id']})
+            else:
+                not_added_counter += 1
```

Comment:
Why would the else condition be met?

Reply:
in case the event is already exists we won't get response from `add_event_response.get('Event')`